### ISSUE TITLE
PPO update schedule

### ISSUE THREAD

**Issue by Ashboy64** [openai/baselines#361]
I'm trying to read through the PPO1 code to gain a better understanding as to how the algorithm works, how one might go about implementing it, etc. I'm confused as to the difference between the "optim_batchsize" and the "timesteps_per_actorbatch" arguments that are fed into the "learn()" function. What are these hyper-parameters? In addition, I see in the "run_atari.py" file, the "make_atari" and "wrap_deepmind" functions are used to wrap the environment. In the "make_atari" function, it uses the "EpisodicLifeEnv", which ends the episode once the a life is lost. On average, I see that the episode length in the beginning of training is about 7 - 8 timesteps, but the batch size is 256, so I don't see how any updates can occur. Thanks in advance for your help.

**Comment by unixpickle** [comment]
I suggest you read the [PPO paper](https://arxiv.org/abs/1707.06347), since it's hard to understand the code without understanding the algorithm on a high-level.

The basic idea is that you have N workers, each with an environment. After all the environments have been stepped for `timesteps_per_actorbatch` steps, all of the experience is put into a dataset, and the network is trained on the dataset using mini-batch sizes determined by `optim_batchsize`.

While an environment is being run, if the episode ends, a new episode is automatically started. So, if you gather 256 timesteps in total but each episode is only 8 timesteps, you will end up with 32 episodes in your buffer.

**Comment by Ashboy64** [comment]
Thanks for the link, I think I understand the algorithm better now. One more question, how are the learning curves shown in the paper generated? More specifically, what does the y-axis represent/ what is the quantity being graphed? Thanks again.

**Comment by unixpickle** [comment]
The x-axis is time, the y-axis is reward. Reward is an environment-specific quantity. For Atari games, it is the score. For MuJoCo, it often indicates a distance of some kind.

**Comment by Ashboy64** [comment]
I see. So after every update to the global policy, it is evaluated by being run in the environment, and the reward after that episode of evaluation is graphed? I'm sorry for the string of questions, I'm a beginner who's just starting to learn more about Reinforcement Learning. Thanks!
